src/main/scala/org/dia/algorithms/mcc/MCCOps.scala

```diff
@@ -207,6 +222,23 @@ object MCCOps {
     return (maxLength, edgeList, hasBorderNode)
   }
 
+  def findConnectedNodes(node: String, edgeMap: mutable.HashMap[String, mutable.Set[String]],
```

Comment:
Add comment string above this function.
Make sure to explain that this is a recursive function and it's termination condition.

Try to write the function tail-recursively, or express it as a loop.
Take a look here for an explanation on tail-recursion:
http://c2.com/cgi/wiki?TailRecursion